### DISCUSSION TITLE
Issue with using heatsource.inp in a situation with multiple dust species

### DISCUSSION THREAD

**Discussion by AuthmaStasley** [dullemond/radmc3d-2.0#56] [General]
Hello, I am a college student working with RADMC3D to model protostellar objects. I have come across a bug in which I get an "ERROR: Too high temperature discovered" when using a heatsource.inp file in conjunction with multiple dust species.

Specifically, heatsource.inp seems to only consider the first species of dust when applying its calculations for dust emission. If species 1 is not present in a cell, even if sufficient dust from other species is present, the simulation raises an error as if we were applying heat to an empty cell.

For a more explicit example to try at home, consider a 1-species dust distribution and a heatsource.inp distribution that reaches out to 10AU, in the form of a generic protostellar disk of some kind. RADMC3D handles this correctly, giving no error if the values are reasonable. However, if I maintain the same parameters, but instead split the dust distribution between two species, one ranging between 0-5AU and the other between 5-10AU (also assuming identical opacities and other parameters; a situation in which the physical setup is exactly identical in practice), we will get a temperature overflow. I have also made sure that there are no accidentally 'empty' cells which may cause an overflow, and indeed, smooth dust distributions will also result in error if the heatsource is not aligned with the first dust species.

This is especially relevant to the current stage of my work, in which I have internal heat emission inside of 10AU, but would like to have regions of different opacities, thus needing different species of dust. I would appreciate if there is some kind of workaround or fix I can implement. I do not understand the underlying code of RADMC3D very well, but it seems like a simple issue of telling the heatsource.inp code which dust species to utilize.

Thank you for your work,

A. Masley

**Comment by vanessa3005** [comment]
Hi A. Masley,

I am currently working on a scenario with multiple dust species as well and get the same error. Did you by any chance find a solution for this issue?

Kind regards,

Vanessa

**Reply by AuthmaStasley** [reply]
Hello Vanessa,

Unfortunately not. The only solution that I found viable was to make species 1 a "grey" opacity species (i.e. with constant, controllable opacity values) and put it wherever heat is injected, and then surround it with your other, more realistic dust species. That way, if there is sufficient optical depth, there should be no difference. If you are not injecting heat into the midplane, or if your system is thin enough that the injected heat is "visible" without absorption and reemission, there may be a problem with this workaround.
